https/tls-config.yml

```diff
@@ -0,0 +1,46 @@
+#TLS CONFIG YAML
+  # Paths to Cert File & Key file from base directory
+  # Both required for valid tls
+  # Paths set as string values
+tlsCertPath : ""
+tlsKeyPath : ""
+
+  # Main config options for tls
+  # Defaults for all options are nil values
+tlsConfig :
+  
+  # Root CA's should be a string path to the set of root certificate authorities
+  # if nil it will use the host's root CA set
+  rootCAs : ~
+  
+  # Server Name used to verify hostname on returned certs
+  # unless Insecure Skip Verify is true
+  serverName : ~ 
+  
+  # Client auth declares the policy the server will follow for client auth
+  # Accepts the following string values and maps to ClientAuth Policies
+  # NoClientCert                - 
+  # RequestClientCert           - 
+  # RequireAnyClientCert        - 
+  # VerifyClientCertIfGiven     - 
+  # RequireAndVerifyClientCert  - 
+  clientAuth : ~ 
+  
+  # Client Ca's accepts a string path to the set of CA's  
+  clientCAs : ~
+
+  # Controls wether a client verifies the servers cert chain and hostname 
+  # Boolean value - TLS insecure if true so should only be set as true for testing
+  insecureSkipVerify : ~ 
+
+  #Array of uint16 supported cipher suites
```

Comment:
The plan of record is not to offer anything below this line, just to stick with Go defaults.

Reply:
@brian-brazil not quite sure I follow your comment. Are you saying a user should not be able to configure cipher suites or prefered cipher suites? Given that Golang supports many insecure ones I think we should expose these options to the users, e.g. to have them prevent down-grade attacks, ... What do you think?

Reply:
Yes, that's what I'm saying. The Go defaults are sane, if someone has a more advanced use case they're free to use a reverse proxy. This was what was agreed at the last dev summit.